app/src/main/java/com/nitramite/courier/ArraPakettiStrategy.java

```diff
@@ -2,88 +2,56 @@
 
 import android.annotation.SuppressLint;
 import android.util.Log;
+
 import com.nitramite.paketinseuranta.EventObject;
+
 import org.json.JSONArray;
 import org.json.JSONException;
 import org.json.JSONObject;
-import java.io.BufferedReader;
+
 import java.io.IOException;
-import java.io.InputStreamReader;
-import java.io.OutputStream;
-import java.net.URL;
-import java.security.KeyManagementException;
-import java.security.NoSuchAlgorithmException;
 import java.text.DateFormat;
 import java.text.SimpleDateFormat;
 import java.util.ArrayList;
 import java.util.Date;
-import javax.net.ssl.HostnameVerifier;
-import javax.net.ssl.HttpsURLConnection;
-import javax.net.ssl.SSLContext;
-import javax.net.ssl.SSLSession;
-import javax.net.ssl.TrustManager;
-import javax.net.ssl.X509TrustManager;
 
-public class ArraPakettiStrategy implements CourierStrategy, HostnameVerifier {
+import okhttp3.MultipartBody;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+
+public class ArraPakettiStrategy implements CourierStrategy {
 
     // Logging
     private static final String TAG = "ArraPakettiStrategy";
 
-    // Host name verifier
-    public boolean verify(String hostname, SSLSession session) {
-        return true;
-    }
-
-
     @Override
     public ParcelObject execute(String parcelCode) {
         ParcelObject parcelObject = new ParcelObject(parcelCode);
         ArrayList<EventObject> eventObjects = new ArrayList<>();
-        TrustManager[] trustAllCerts = new TrustManager[]{
-                new X509TrustManager() {
-                    public java.security.cert.X509Certificate[] getAcceptedIssuers() {
-                        return null;
-                    }
-                    @SuppressLint("TrustAllX509TrustManager")
-                    public void checkClientTrusted(
-                            java.security.cert.X509Certificate[] certs, String authType) {
-                    }
-                    @SuppressLint("TrustAllX509TrustManager")
-                    public void checkServerTrusted(
-                            java.security.cert.X509Certificate[] certs, String authType) {
-                    }
-                }
-        };
         try {
-            SSLContext sc = SSLContext.getInstance("SSL");
-            sc.init(null, trustAllCerts, new java.security.SecureRandom());
-            HttpsURLConnection.setDefaultSSLSocketFactory(sc.getSocketFactory());
             String url = "https://www.r-kioski.fi/wordpress/wp-admin/admin-ajax.php";
```

Comment:
Unrelated but maybe add this as a constant for tidyness?

Reply:
Like this? https://github.com/norkator/Paketin-Seuranta/pull/18/commits/c2c3f429fc836d6a33ff68eddf661ee0a26ef0a0
I left those intact which parts are constructed like url + tracking number + some latter part